024-adopt-the-kafka-static-quota-plugin.md

```diff
@@ -0,0 +1,63 @@
+# Adopt the Kafka Static Quota plugin
+
+## Current situation
+
+One of the common problems when running Apache Kafka is running out of disk space.
+Strimzi currently does not offer any protection against this.
+Once the disk gets full, usually the only way how to recover from full disk is to either delete some files (log segments) or to increase the disk size.
+
+## Motivation
+
+Having some solution which would protect from the disks getting full would be a good improvement for Strimzi users.
+
+## Proposal
+
+Strimzi should adopt the [Kafka Static Quota plugin](https://github.com/lulf/kafka-static-quota-plugin).
+It provides two types of quotas:
+
+### Per-broker produce and fetch quotas
+
+Apache Kafka itself offers only quotas per client / user.
+The Kafka Static Quota plugin allows configuring of an overall produce and/or fetch quota per-broker.
+The quota will be distributed between the clients connected to the broker.
+The produce / fetch quotas currently don't support per-listener quotas.
+There is always only one quota for all listeners.
+The replication traffic is not counted into the quota.
+
+### Storage quotas
+
+The Kafka Static Quota plugin allows users to configure two storage limits: soft and hard.
+After the soft limit is breached, it will start throttling the producers.
+The allowed throughput is linearly decreased until the hard limit is reached.
+Once the hard limit is reached, the allowed produce throughput will be 0.
+The storage quotas currently don't support JBOD storage, only brokers with single log directory.
+
+### Plugin adoption
```

Comment:
This should be ##

Reply:
Why do you think so? It is.a description of what I propose. So it IMHO should be in the `## Proposal` section.

Reply:
Previously you mentioned "two types of quotas" and used ### for them. So it's weird having another ### section after them because it's not a type of quotas.